### ISSUE TITLE
Determine folder age recursively

### ISSUE THREAD

**Issue by bibermann** [SebastianMC/obsidian-custom-sort#132]
Hello,

this Plugin is really nice. The main reason why I was looking for such a plugin was that folders were not sorted by date. So I was glad to find the "advanced modified" sort order.

However, I'd like the function `determineDatesForFolder` to go recursively into sub directories, so that also parent nodes are pushed to the top when I change a deeply nested note. This behavior could be a distinct sort order, e.g. "recursive modified".

Is there a reason why we are only looking at the first level currently?

**Comment by SebastianMC** [comment]
Hi @bibermann 

the plugin release [2.1.8](https://github.com/SebastianMC/obsidian-custom-sort/releases/tag/2.1.8) contains the feature which you asked for.

The names of new orders are `advanced recursive created` and `advanced recursive modified`

Introduced under the disclaimer 'experimental feature' - use with care, for extremely broad and deep vaults the overhead could be noticeable on slower mobile devices

Let me know if it works on your side

**Comment by bibermann** [comment]
> Hi @bibermann
> 
> the plugin release [2.1.8](https://github.com/SebastianMC/obsidian-custom-sort/releases/tag/2.1.8) contains the feature which you asked for.
> 
> The names of new orders are `advanced recursive created` and `advanced recursive modified`
> 
> Introduced under the disclaimer 'experimental feature' - use with care, for extremely broad and deep vaults the overhead could be noticeable on slower mobile devices
> 
> Let me know if it works on your side

Works like a charm, thank you very much!

Is your statement "experimental feature" explaining why you've not added this to the documentation yet?

**Comment by SebastianMC** [comment]
Hi @bibermann 

regarding the documentation ;-) it is (very) outdated now.
In the sense that it doesn't describe most the recent features added in the past year or so.
I only managed to extend the documentation with the bookmarks plugin integration feature.
The main reason for the docs not being up-to-date is their format - it was just recently that I decided to migrate and update the documentation to a new very convenient format (for the reader and also for the author).
I borrowed the idea of incorporating _Material for MkDocs_ for documentation from the well-knowns Obsidian dataview plugin. There is a fresh new [documentation project](https://github.com/SebastianMC/obsidian-custom-sort-docs) started a month ago with the intention to migrate and update the documentation. So this is the big picture.

The meaning of "experimental feature": this label is used for a feature, which is not settled down yet, needs time for real-life testing by users, can then evolve in a more advanced or mature form or be replaced with an equivalent feature.
